### PR TITLE
Display errors in web dashboard, clean up and reorganize CSS/Classes

### PR DESCRIPTION
**Pull request by flodolo**
- Refactor and cleanup classes: move Bugzilla functions in their own class, remove access to global constants in Utils
- Fix warnings for son/sat (missing Bugzilla references)
- Display different table header for lang/raw files
- Improve CSS at smaller resolutions, clean up over qualified rules, fix background disappearing when scrolling horizontally on larger pages
- Clean up phpDOC syntax in classes

### CODE REVIEW

File: models/locale.php [hunk 1]
```diff
@@ -4,10 +4,10 @@
  */
 namespace Webdashboard;
 
-$json   = isset($_GET['json']) ? true : false;
+$json = isset($_GET['json']) ? true : false;
```
Comment (pascalchevrel) [inline models/locale.php]: we can simplify here with:
```php
$json = isset($_GET['json']);

File: classes/Webdashboard/Json.php [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+    private $jsonURI;
+
+    /**
+     * Build Json object and set URI (default empty)
+     *
+     * @param  string  $uri  URI of the JSON stream to read
+     */
+    public function __construct($uri = '')
+    {
+        $this->jsonURI = $uri;
```
Comment (pascalchevrel) [inline classes/Webdashboard/Json.php]: since this is no longer a static function, you can return the object in your setter:
```php
return $this;
```
This way you can chain the outputContent method to the setter (I'll put an example in the model where this is used)

File: models/locale.php [hunk 3]
```diff
@@ -19,19 +19,22 @@
     $locale = $_GET['locale'];
 }
 
-// get lang files status from langchecker
-$lang_files = Json::fetch(LANG_CHECKER . "?locale={$locale}&json");
+// Get lang files status from langchecker
+$json_data = new Json();
+$json_data->setURI(LANG_CHECKER . "?locale={$locale}&json");
+$lang_files = $json_data->fetchContent();
```
Comment (pascalchevrel) [inline models/locale.php]: You could use method chaining which I think makes the code easier to read:
```php
$json_data = new Json();
$lang_files = $json_data
    ->setURI(LANG_CHECKER . "?locale={$locale}&json")
    ->fetchContent();
```

File: classes/Webdashboard/Utils.php [hunk 4]
```diff
@@ -1,52 +1,28 @@
...
-     * @return string
+     * @param   string  $url         Content to fetch
+     * @param   string  $time        Seconds to cache (default 120)
+     * @param   string  $cache_path  Path to cache
+     *
+     * @return  string               URI of the cached resource, or original
+     *                               source if not available
      */
-    public static function cacheUrl($url, $time = 120)
+    public static function cacheUrl($url, $time = 120, $cache_path)
```
Comment (pascalchevrel) [inline classes/Webdashboard/Utils.php]: We do have a CACHE constant on the Webdashboard that is in use and making that a $cache_path variable means that we now add CACHE as a target to all the uses of cacheUrl(). Since this class is a simple cache system with just static method, I don't think it makes code easier. Do you have plan to cache data outside of the current cache folder?

Comment (flodolo) [inline classes/Webdashboard/Utils.php]: The idea is to not have a class method depending on something (in this case a global constant) that's not defined or available in the class itself. It makes code harder to understand, and in this case the cost is only to have an additional parameter in method calls. 

In the long term it probably makes sense to have a Cache object where the cache path is defined only once for istance.

File: models/locale.php [hunk 5]
```diff
@@ -19,19 +19,22 @@
     $locale = $_GET['locale'];
 }
 
-// get lang files status from langchecker
-$lang_files = Json::fetch(LANG_CHECKER . "?locale={$locale}&json");
+// Get lang files status from langchecker
+$json_data = new Json();
```
Comment (pascalchevrel) [inline models/locale.php]: FYI, since you don't pass any parameter to the constructor, you could omit parenthesis:
```php
$json_data = new Json;
```

File: models/project.php [hunk 6]
```diff
@@ -2,13 +2,15 @@
...
 
 // Base for the query to get external data
 $langchecker_query = LANG_CHECKER . '?locale=all&json';
 $locale_done = [];
 
-// include all data about project pages
+// Include all data about project pages
 include __DIR__ . '/../data/project.php';
 
 $requested_project = (array_key_exists($_GET['project'], $projects)) ? $_GET['project'] : 'default';
```
Comment (pascalchevrel) [inline models/project.php]: could you remove the parenthesis around array_key_exists()? Thanks

File: classes/Webdashboard/Utils.php (hunk 4 above)
Comment (pascalchevrel) [inline classes/Webdashboard/Utils.php]: Well, personnally I think that adding the same parameter to all the calls to cache_url is worse than having an external constant in the class, especially since this is a utility class with static methods. If we need to worry about that, then I think we should have a real object and define the cache folder in the constructor, or see if we can reuse the cache library from transvision but that's maybe overkill.

File: models/locale.php (hunk 3 above)
Comment (flodolo) [inline models/locale.php]: Interesting. The downside is that I'm forced to create a new object if I need to fetch multiple feeds in the same view, while now I just create one instance in the file, then update the URI and fetch.

Should I switch anyway to multiple instances?

Comment (pascalchevrel) [inline models/locale.php]: you don't need to create a new object, you can do that:
```php
// Get lang files status from langchecker
$json_data = new Json;
$lang_files = $json_data
->setURI(LANG_CHECKER . "?locale={$locale}&json")
->fetchContent();

// Check if the locale is working on locamotion
$locamotion = $json_data
    ->setURI(Utils::cacheUrl(LANG_CHECKER . '?action=listlocales&project=locamotion&json', 15*60, CACHE))
    ->fetchContent();
```

File: classes/Webdashboard/Utils.php (hunk 4 above)
Comment (flodolo) [inline classes/Webdashboard/Utils.php]: Wondering: what about setting the default to CACHE and clean up the calls?

File: models/locale.php (hunk 3 above)
Comment (flodolo) [inline models/locale.php]: Tried but it didn't work. Let me try again, maybe I just screwed up something silly.

**Comment by flodolo** [comment]
No idea what I did wrong with the object before :8ball: 

Can you take a look at the commit and tell me if it makes sense before squashing?
https://github.com/flodolo/webdashboard/commit/a95185dcb8fc9dfe275c1d8bdda88bb3bd520b31
